### PR TITLE
data_prep: use argparse for introspector

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Use argparse for introspector's module. This fixes a todo and will also make it easier to add more feature to the introspector module.

### CODE REVIEW

**Comment by DonggeLiu** [comment]
> seems to be unrelated to the changes?

Yes, it is unrelated.
That CI builds a docker image based on the code in the PR and pushes it to an artifact registry so that we can run experiments with it.

File: data_prep/introspector.py [hunk 1]
```diff
@@ -452,32 +453,46 @@ def get_project_funcs(project_name: str) -> Dict[str, List[Dict]]:
   return fuzz_target_funcs
 
 
+def _parse_arguments() -> argparse.Namespace:
+  """Parses command line args."""
+  parser = argparse.ArgumentParser(
+      description='Parse arguments to generate benchmarks')
```
Comment (DonggeLiu) [inline data_prep/introspector.py:459]: Nitpicking: For better consistency, could you please end the description with a period (e.g., `.`)? Thanks!
Same below.

File: data_prep/introspector.py [hunk 2]
```diff
@@ -452,32 +453,46 @@ def get_project_funcs(project_name: str) -> Dict[str, List[Dict]]:
...
-  max_num_function = 3
-  if len(sys.argv) > 2:
-    max_num_function = int(sys.argv[2])
-  if len(sys.argv) > 3:
-    outdir = sys.argv[3]
-    os.makedirs(outdir, exist_ok=True)
-  else:
-    outdir = ''
+  args = _parse_arguments()
+  if args.out != '':
```
Comment (DonggeLiu) [inline data_prep/introspector.py:481]: Nitpicking: How about `if args.out:` as we have specified its type to `str` above?

**Review by DonggeLiu — APPROVED** [review]
Thanks so much for the PR, @DavidKorczynski !
I left some minor nitpicking comments, hope you would not mind :)

File: data_prep/introspector.py (hunk 2 above)
Comment (DavidKorczynski) [reply data_prep/introspector.py]: Sounds good, done!

File: data_prep/introspector.py (hunk 1 above)
Comment (DavidKorczynski) [reply data_prep/introspector.py]: Sounds good, done!

**Review by DonggeLiu — APPROVED** [review]
Thanks again, @DavidKorczynski !

**Pull request merged**